docs/proposal/20220929-ipam-support.md

```diff
@@ -0,0 +1,423 @@
+   to be updated to accept this IP address(es) and populate the addresses field
+   with the IPs for the corresponding device.
+
+   The device index on the IPAddressClaim name will be used to
+ configure the
+   IPAddress on the correct device.
+
+1. Update the VSphereMachine status to reflect the network device state. Update
+   the IPAddressClaim condition.
+
+    ```yaml
+    ---
+    status:
+      conditions:
+        - lastTransitionTime: "2022-09-14T14:00:00Z"
+          type: IPAddressClaimed
+          status: "True"
+      network:
+        - connected: true
+          ipAddrs:
+            - 10.10.10.100
+          macAddr: 00:01:02:03:04:05
+          networkName: VM Network
+    ```
+
+#### Owner Reference
+
+When a VSphereMachine is deleted, an existing finalizer prevents the removal of
+the VSphereMachine object until the VM in vSphere is gone. As long as the
+IPAddressClaim has an owner reference to the VSphereMachine it will get safely
+deleted when the VSphereMachine is deleted. The deletion of the IPAddressClaim
+when the VSphereMachine is deleted is safe because the VM is deleted and is no
+longer using the IP.
+
+#### Flavors
+
+An IPAM flavor will be added to support configuring an array of group, kind,
+name, VM Network objects. When the IPAM flavor is specified, the devices in the
+network spec of the Nodes will have DHCP turned off. Each of the device's
+fromPool in the network spec will be populated with one of the above mentioned
+configuration objects. The templating and configurability of flavors may
+restrict the possible ways IPAM can be configured through flavors.
+
+#### Validation
+
+In the NetworkDeviceSpec:
+
+The IPAddrs field is required if DHCP4 or DHCP6 is false. The new fromPools
+field allows for a third option and therefore we need to update validation to
+ensure one of the three options is specified for a network device.
+
+#### Nameservers
+
+Nameservers can come from DHCP if enabled. Therefore when DHCP is disabled and
+the `FromPools` field is set  a user will have to supply their own nameservers
+in the device spec.
+
+### Security Model
+
+- Following the original IPAM integration proposal, the VSphereMachine is
+  required to be in the same namespace as the IPAddressClaim, IPAddress, and
+  IPPool.
+  - For IPAddressClaim and IPAddress this is enforced by the lack of a
+    namespace field on the reference
+  - The fromPools field on the VSphereMachineTemplate/VSphereMachine also
+    lacks a namespace field on the reference.
+
+### Risks and Mitigations
+
+- Unresponsive IPAM providers can prevent successful creation of a cluster
+- Any issues with fulfilling the IPAddressClaim can prevent successful creation of a cluster
+  - This is partially by design i.e we don't want the cluster to be created if
+    it can't fulfill the ip address claim, but this is just added complexity
+    that a user has to be aware of.
+- Wrong IP address allocations (e.g. duplicates) can brick a cluster’s network

```

Comment:
A nit pick: brick->break. Or does this have a meaning I am not aware of?

Reply:
IT slang for a broken device. A device without its firmware/software and no way to recover it has no use, but you can still use it as a brick or paperweight.
It's used as a synonym for "break" here. Might be my fault if it's from the CAPI IPAM proposal.